### PR TITLE
Add DAP Intro Option for Node Certificate Rotation

### LINKED ISSUE #89: DAP Intro CLI tool supports certificate rotation
As a Conjur developer
I want the `bin/dap` tool to support certification rotation
So that I can quickly test this procedure with a DAP cluster

### PR DESCRIPTION
**Pull request by micahlee**
This PR modifies DAP intro in a couple of ways:

* It updates the HAProxy config to support older versions of DAP by using the standard HTTPS port (443) for health checks, rather than rely on the newer HTTP health check on port 444.

* It adds a procedure flag to dap intro to rotate the node certificates called `--rotate-custom-certificates`. This allows for quickly testing certificate rotation for the Master, Standbys, and Follower in DAP Intro.

Closes #89

To try the new rotation function:

```
# Provision a Master
bin/dap --provision-master

# Import custom certificates
bin/dap --import-custom-certificates

# Provision Standbys
bin/dap --provision-standbys

# (Optional) Enable Auto-failover
bin/dap --enable-auto-failover

# Provision Follower
bin/dap --provision-follower

# Verify the deployment is working
bin/api --load-policy-and-values
bin/api --fetch-secrets

# Rotate the node certificates
bin/dap --rotate-custom-certificates

# Verify the deployment is still working
bin/api --load-policy-and-values
bin/api --fetch-secrets
```

### CODE REVIEW

File: artifacts/certificate-generator/generate-certs-script [hunk 1]
```diff
@@ -44,52 +52,54 @@ create_folders() {
...
+      # Sign Intermediate with Root
+      cfssl sign \
+        -ca $root_destination.pem \
+        -ca-key $root_destination-key.pem \
+        -config $config_path/cfssl.conf \
+        -profile intermediate_ca \
+        $intermediate_destination.csr | cfssljson -bare $intermediate_destination
+    fi
+
+    cat $certificate_path/root_ca/ca.pem \
```
Comment (h-artzi) [inline artifacts/certificate-generator/generate-certs-script:77]: This backwards from the convention. It should be intermediate, then root.

File: bin/dap [hunk 2]
```diff
@@ -299,9 +299,95 @@ function _import_certificates {
...
+  
+    # Import the new Follower certificate into each DAP Standby so that it
+    # is available through the seed service if the Standby is promoted.
+    _run conjur-master-3.mycompany.local \
+      "evoke ca import --force --key $cert_path/dap_follower/dap-follower-key.pem $cert_path/dap_follower/dap-follower.pem"
+  fi
+
+  # Update the Postgres replication config in the standbys to use the rotated
+  # certificates and re-establish replication.
+  #_rebase_standby_replication
```
Comment (h-artzi) [inline bin/dap:347]: Is this commented out on purpose?

**Review by h-artzi — CHANGES_REQUESTED** [review]

File: artifacts/certificate-generator/generate-certs-script (hunk 1 above)
Comment (micahlee) [reply artifacts/certificate-generator/generate-certs-script:77]: Interesting. That's not a new change, I just moved where it happens

I'll change it though and make sure everything still works as expected. Thanks!

File: bin/dap (hunk 2 above)
Comment (micahlee) [reply bin/dap:347]: Nope, I need to remove that. Thanks!

File: artifacts/certificate-generator/generate-certs-script (hunk 1 above)
Comment (h-artzi) [reply artifacts/certificate-generator/generate-certs-script:77]: I currently have a PR open on dap-intro that addresses this "issue", but I'll close it once this is merged!

**Comment by micahlee** [comment]
Alright, @h-artzi, this should be ready for another look. Thanks!

**Review by h-artzi — APPROVED** [review]
LGTM!

**Pull request merged**